x/ccv/provider/keeper/key_assignment.go

```diff
@@ -15,6 +16,53 @@ import (
 	ccvtypes "github.com/cosmos/interchain-security/v4/x/ccv/types"
 )
 
+// ParseConsumerKey parses the ED25519 PubKey`consumerKey` from a JSON string
+// and constructs its corresponding `tmprotocrypto.PublicKey`
+func (k Keeper) ParseConsumerKey(consumerKey string) (tmprotocrypto.PublicKey, error) {
```

Comment:
Took this method code _as is_ from `msg_server.go` (see below). I chose not to reuse the `AssignConsumerKey` method in `partial_set_security.go` when opting in a validator with a consumer key because this would entail:
* creating a `MsgAssign` message inside `HandleOptIn` which seems counterintuitive in the sense that a message creates another message internally to do something;
*  `AssignConsumerKey` would emit emit an assignment event and potentially this could confuse relayers becuse for one `MsgOptIn` message, relayers get events as if another message was executed as well.